tests/integration/ha_tests/helpers.py

```diff
@@ -504,7 +512,28 @@ async def db_step_down(ops_test: OpsTest, unit_ip: str):
             continue
 
         # this message indicates that the previous primary performed a repl step down operation.
-        if item["msg"] == "Starting an election due to step up request":
+        # its important to check that this step down was performed after the sigterm opteration
+        # was performed, as it could have been performed at an earlier time for another reason.
+        step_down_time = convert_time(item["t"]["$date"])
+        if (
+            item["msg"] == "Starting an election due to step up request"
+            and step_down_time > sigterm_time
+        ):
             return True
 
     return False
+
+
+def convert_time(time_as_str: str) -> int:
+    """Converts a string time representation to an integer time representation."""
+    # parse time representation, provided in this format: 'YYYY-MM-DDTHH:MM:SS.MMM+00:00'
+    y_m_d, h_m_s_m = time_as_str.split("T")
+    year, month, day = y_m_d.split("-")
+    hour, minute, second = h_m_s_m.split(".")[0].split(":")
+    microsecond = h_m_s_m.split(".")[1].split("+")[0]
+
+    d = datetime.datetime(
+        int(year), int(month), int(day), int(hour), int(minute), int(second), int(microsecond)
+    )
```

Comment:
You can get your `datetime` object with `strptime`:
```python
from datetime import datetime

a = "2022-08-09T09:15:28.131+00:00"   # Mongo log

d = datetime.strptime(b, "%Y-%m-%dT%H:%M:%S.%f%z")
```
You can check the reference for format codes in [the docs](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes), in case I made some mistake :)

Reply:
soooo much nicer!! 🤩🤩🤩 thank you so much for sharing :)